scripts/sdp/clean_asm.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+import os
+import sys
+
+def clean_assembly_file(input_file):
+    """Cleans an assembly file by removing lines starting with # and saves the changes in the same file."""
```

Comment:
I don't delete these lines because from what I found `#APP` means that the following code was written by the user, rather than generated by the compiler.
`#NO_APP` means that the following code was written by the compiler.
In some cases the GNU assembler uses a different approach to parsing code generated by the compiler (i.e., after #NO_APP), because it knows that the white space will be very regular and some types of code will never appear.

Reply:
> and saves the changes in the same file.

when something fails, then debugging becomes much harder if overwriting / modifying generated files, as the original file is suddenly lost.